fix(formatter): should not insert a semicolon for non-property signatures when no semicolon is set

Dunqing opened the pull request:
Prettier issue: https://github.com/prettier/prettier/issues/14040

Only property signatures would conflict with the following signatures in some situations, so return early if the current signature is not a property when `no-semi` is set.

leaysgur commented:
### Merge activity

* **Dec 11, 12:52 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 11, 12:52 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Dunqing committed:
fix(formatter): should not insert a semicolon for non-property signatures when no semicolon is set (#16700)

Prettier issue: https://github.com/prettier/prettier/issues/14040

Only property signatures would conflict with the following signatures in some situations, so return early if the current signature is not a property when `no-semi` is set.
